can/broadcastmanager.py

```diff
@@ -34,28 +35,43 @@ class CyclicSendTaskABC(CyclicTask):
     Message send task with defined period
     """
 
-    def __init__(self, message, period):
+    def __init__(self, messages, period):
         """
-        :param can.Message message: The message to be sent periodically.
-        :param float period: The rate in seconds at which to send the message.
+        :param List[can.Message] messages: The message to be sent periodically.
+        :param float period: The rate in seconds at which to send the messages.
         """
-        self.message = message
-        self.can_id = message.arbitration_id
-        self.arbitration_id = message.arbitration_id
+        if not isinstance(messages, list):
```

Comment:
TBH I'm not really a fan of keeping backwards compatibility and supporting both a `List` and a `Message` type. It seems kludgy to muddle the API like that. Plus, a single Message can still be supported by a list of length 1...

Reply:
No one has really requested this feature for years and neither one of IXXAT, Kvaser, NI-CAN, or Systec interfaces which provide cyclic message transmissions have seen the need for this and won't support it. Therefore I believe it is not justified to force everyone to change their implementations for the very few people that need this. At the very least we must have a deprecation period to allow implementations to adjust. A library that changes its API in backwards incompatible ways too often are not very popular among developers relying on it for critical applications.

Maybe it only has to be converted to a list at one place. Maybe in the `send_periodic()` method?